src/hotspot/share/runtime/osThread.cpp

```diff
@@ -38,18 +38,18 @@ OSThread::~OSThread() {
 
 // Printing
 void OSThread::print_on(outputStream *st) const {
-  st->print("nid=0x%x ", thread_id());
+  st->print("nid=%d ", thread_id());
```

Comment:
thread_id is of an opaque type (eg pthread_t). I think we can reasonably assume its numeric, but I would print it as an unsigned 64bit int just in case.

Reply:
Hi Thomas, we can not use other format specifiers (`%ld`,`%llu`) after my practice, because it can not compile on my mac:
```
/home/qingfeng.yy/jdktip/src/hotspot/share/runtime/osThread.cpp:41:21: error: format '%llu' expects argument of type 'long long unsigned int', but argument 3 has type 'OSThread::thread_id_t' {aka 'int'} [-Werror=format=]
   41 |   st->print("nid=%llu ", thread_id());
      |                  ~~~^    ~~~~~~~~~~~
      |                     |             |
      |                     |             OSThread::thread_id_t {aka int}
      |                     long long unsigned int
      |                  %u

/home/qingfeng.yy/jdktip/src/hotspot/share/runtime/osThread.cpp:41:20: error: format '%ld' expects argument of type 'long int', but argument 3 has type 'OSThread::thread_id_t' {aka 'int'} [-Werror=format=]
   41 |   st->print("nid=%ld ", thread_id());
      |                  ~~^    ~~~~~~~~~~~
      |                    |             |
      |                    long int      OSThread::thread_id_t {aka int}
      |                  %d
cc1plus: all warnings being treated as errors
```
After a quick search, I find many occurrences that using `%d` as a format specifier, so it looks ok.

(Conversions looks redundant)
https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/interpreter/bytecodeTracer.cpp#L100

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/interpreter/bytecodeTracer.cpp#L114

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/deoptimization.cpp#L838

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/safepoint.cpp#L892-L894

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/safepointMechanism.cpp#L102-L113

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/stackWatermark.cpp#L114-L115

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/stackWatermark.cpp#L199-L200

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/stackWatermark.cpp#L238-L239

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/thread.cpp#L619

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/runtime/thread.cpp#L2098

https://github.com/openjdk/jdk/blob/f45be1519e84dd0562a841ceb23bcec4dec1437f/src/hotspot/share/utilities/vmError.cpp#L763

Reply:
You'd do:
```
print("nid: " UINT64_FORMAT, (uint64_t) id):;
```
thread_t is, among other things, pthread_t, which is opaque. Any current code treating that as signed int is incorrect too.